src/chess/pgn.h

```diff
@@ -75,31 +77,29 @@ class PgnReader {
     bool in_comment = false;
     bool started = false;
     while (GzGetLine(file, line)) {
+      line_no_++;
       // Check if we have a UTF-8 BOM. If so, just ignore it.
       // Only supposed to exist in the first line, but should not matter.
       if (line.substr(0,3) == "\xEF\xBB\xBF") line = line.substr(3);
       if (!line.empty() && line.back() == '\r') line.pop_back();
       // TODO: support line breaks in tags to ensure they are properly ignored.
-      if (line.empty() || line[0] == '[') {
+      if ((line.empty() || line[0] == '[') && !in_comment) {
         if (started) {
-          Flush();
+          Flush("*");
```

Comment:
Probably `started_=false` makes sense to move to `Flush()`?
(also maybe rename `Flush()` to something like `AppendCurrentGame()` or at least `Finalize()`? Especially now that it takes a std::string parameter, it's not just flushing buffer.